Packs/ApiModules/Scripts/SiemApiModule/SiemAPIModule.py

```diff
@@ -0,0 +1,161 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+
+from abc import ABC
+from typing import Any, Optional
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+from enum import Enum
+from pydantic import BaseConfig, BaseModel, AnyUrl, validator  # type: ignore[E0611, E0611, E0611]
+from requests.auth import HTTPBasicAuth
+
+
+class Method(str, Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    HEAD = 'HEAD'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+def load_json(v: Any) -> dict:
+    if not isinstance(v, (dict, str)):
+        raise ValueError('headers are not dict or a valid json')
+    if isinstance(v, str):
+        try:
+            v = json.loads(v)
+            if not isinstance(v, dict):
+                raise ValueError('headers are not from dict type')
+        except json.decoder.JSONDecodeError as exc:
+            raise ValueError('headers are not valid Json object') from exc
+    if isinstance(v, dict):
+        return v
+
+
+class IntegrationHTTPRequest(BaseModel):
+    method: Method
+    url: AnyUrl
+    verify: bool = True
+    headers: dict = dict()  # type: ignore[type-arg]
+    auth: Optional[HTTPBasicAuth]
+    data: Any = None
+
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    _normalize_headers = validator('headers', pre=True, allow_reuse=True)(
+        load_json
+    )
+
+
+class Credentials(BaseModel):
+    password: str
+    identifier: Optional[str]
+
+
+def set_authorization(request: IntegrationHTTPRequest, auth_credendtials):
+    """Automatic authorization.
+    Supports {Authorization: Bearer __token__}
+    or Basic Auth.
+    """
+    creds = Credentials.parse_obj(auth_credendtials)
+    if creds.password and creds.identifier:
+        request.auth = HTTPBasicAuth(creds.identifier, creds.password)
+    auth = {'Authorization': f'Bearer {creds.password}'}
+    if request.headers:
+        request.headers |= auth  # type: ignore[assignment, operator]
+    else:
+        request.headers = auth  # type: ignore[assignment]
+
+
+class IntegrationOptions(BaseModel):
+    """Add here any option you need to add to the logic"""
+
+    proxy: bool = False
+    limit: int = 1000
+
+
+class IntegrationEventsClient(ABC):
```

Comment:
why aren't we using the BaseClient? it already has some error handling features that are lacking here

Reply:
because it is breaking the requests API and is very hard to use in comparison to just sending a request.
also, we are missing errors because of that.